byway/src/main/webapp/script.js

```diff
@@ -12,17 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+var selected = [];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/** Track the selected interests in an array according to their status. */
+function checkStatus(elem) {
+    if(elem.className === "btnActive") {
+        selected.push(elem.innerText);
+    } else {
+        selected = selected.filter(function(interest) {
+            return interest !== elem.innerText;
+        });
+    }
+}
+
+/** Update the class of the buttons to change their style. */
+function switchStatus(elem) {
+    var switchClass = (elem.className === "btn"? "btnActive": "btn");
+    elem.className = switchClass;
+    checkStatus(elem);
+}
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+/** Display all the buttons onscreen with independent onClick events. */
+function loadButtons() {
+    var buttonSection = document.getElementById("interests");
+    fetch('/places').then(response => response.json()).then((places) => {
+        for(var i = 0; i < places.length; i++) {
+            var button = document.createElement("button");
+            button.innerHTML = places[i];
```

Comment:
i think innerText is generally recommended over innerHTML

Reply:
Didn't notice I used innerHTML instead - thanks for catching! Fixed